### PR TITLE
OAuth api printers

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

**Comment by liggitt** [comment]
@deads2k review

**Comment by smarterclayton** [comment]
LGTM [merge]

**Comment by deads2k** [comment]
Add describers while you're here.

File: pkg/cmd/cli/describe/printer.go [hunk 1]
```diff
@@ -29,6 +31,14 @@ var (
 	parameterColumns        = []string{"NAME", "DESCRIPTION", "GENERATOR", "VALUE"}
 	policyColumns           = []string{"NAME", "ROLES", "LAST MODIFIED"}
 	policyBindingColumns    = []string{"NAME", "ROLE BINDINGS", "LAST MODIFIED"}
+
+	oauthClientColumns              = []string{"NAME", "SECRET", "REDIRECT URIS"}
+	oauthClientAuthorizationColumns = []string{"NAME", "USER NAME", "CLIENT NAME", "SCOPES"}
+	oauthAccessTokenColumns         = []string{"NAME", "USER NAME", "CLIENT NAME", "CREATED", "EXPIRES", "REDIRECT URI", "SCOPES"}
```
Comment (deads2k) [inline pkg/cmd/cli/describe/printer.go]: lack of refresh token intentional?

Comment (liggitt) [inline pkg/cmd/cli/describe/printer.go]: we don't support refresh token at this point, so yes

**Comment by liggitt** [comment]
describers would require a client, which we don't have yet. I have a TODO for clients

File: pkg/cmd/cli/describe/printer.go [hunk 2]
```diff
@@ -29,6 +31,14 @@ var (
...
+	oauthAuthorizeTokenColumns      = []string{"NAME", "USER NAME", "CLIENT NAME", "CREATED", "EXPIRES", "REDIRECT URI", "SCOPES"}
```
Comment (deads2k) [inline pkg/cmd/cli/describe/printer.go]: Describer could add State.

File: pkg/cmd/cli/describe/printer.go [hunk 3]
```diff
@@ -29,6 +31,14 @@ var (
 	parameterColumns        = []string{"NAME", "DESCRIPTION", "GENERATOR", "VALUE"}
 	policyColumns           = []string{"NAME", "ROLES", "LAST MODIFIED"}
 	policyBindingColumns    = []string{"NAME", "ROLE BINDINGS", "LAST MODIFIED"}
+
+	oauthClientColumns              = []string{"NAME", "SECRET", "REDIRECT URIS"}
```
Comment (deads2k) [inline pkg/cmd/cli/describe/printer.go]: Since there's no describer, can you add RepondWithChallenges?

**Pull request merged**